specifics/Dhyana/DhyanaStateMachine.cpp

```diff
@@ -1,212 +1,472 @@
-//
-//  		This file is generated by POGO
-//	(Program Obviously used to Generate tango Object)
+// $HeadURL:  $
 //
-//         (c) - Software Engineering Group - ESRF
 //=============================================================================
-
-#ifdef WIN32
-#include "tango.h"
-#endif
+//                This file is generated by POGO
+//        (Program Obviously used to Generate tango Object)
+//=============================================================================
 
 #include <Dhyana.h>
-#include <DhyanaClass.h>
 
-#ifndef WIN32
-#include "tango.h"
-#endif
+/*----- PROTECTED REGION END -----*/	//	Dhyana::DhyanaStateMachine.cpp
 
+//================================================================
+//  States   |  Description
+//================================================================
+//  FAULT    |  
+//  INIT     |  
+//  RUNNING  |  
+//  STANDBY  |  
 
-/*====================================================================
- *	This file contains the methods to allow commands and attributes
- * read or write execution.
- *
- * If you wand to add your own code, add it between 
- * the "End/Re-Start of Generated Code" comments.
- *
- * If you want, you can also add your own methods.
- *====================================================================
- */
 
 namespace Dhyana_ns
 {
-
 //=================================================
 //		Attributes Allowed Methods
 //=================================================
 
-
-//+----------------------------------------------------------------------------
-//
-// method : 		Dhyana::is_tucamVersion_allowed
-// 
-// description : 	Read/Write allowed for tucamVersion attribute.
-//
-//-----------------------------------------------------------------------------
-bool Dhyana::is_tucamVersion_allowed(Tango::AttReqType type)
+//--------------------------------------------------------
+/**
+ *	Method      : Dhyana::is_tucamVersion_allowed()
+ *	Description : Execution allowed for tucamVersion attribute
+ */
+//--------------------------------------------------------
+bool Dhyana::is_tucamVersion_allowed(TANGO_UNUSED(Tango::AttReqType type))
 {
-	if (get_state() == Tango::INIT	||
-		get_state() == Tango::FAULT	||
-		get_state() == Tango::RUNNING)
+
+	//	Check access type.
+	if ( type==Tango::READ_REQ )
 	{
-		//	End of Generated Code
-		if ( get_state()==Tango::RUNNING && type==Tango::READ_REQ )
+		//	Compare device state with not allowed states for READ 
+		if (get_state()==Tango::INIT ||
+			get_state()==Tango::FAULT ||
+			get_state()==Tango::RUNNING)
 		{
-           return true;
-		}
+		/*----- PROTECTED REGION ID(Dhyana::tucamVersionStateAllowed_READ) ENABLED START -----*/
 		
-		if ( get_state()==Tango::FAULT && is_device_initialized() )
-		{
-           return true;
-		}		
-		//	Re-Start of Generated Code
-		return false;
+		/*----- PROTECTED REGION END -----*/	//	Dhyana::tucamVersionStateAllowed_READ
+			return false;
+		}
+		return true;

```

Comment:
The state machine is wrong (because of Pogo 9): please see with Arafat to manage it together